### PR TITLE
fix: reject non-object workflow caches

### PR DESCRIPTION
**Pull request by marcelsafin**
## Summary

- treat non-object workflow and step cache metadata as invalid
- refetch when a cached workflow catalog is not a JSON object
- reject non-object stale cache fallbacks instead of returning unusable data

## Testing

- `uvx ruff@0.15.0 check src tests`
- `.venv/bin/python -m pytest -q tests/test_workflows.py` (821 passed)
- `.venv/bin/python -m pytest -q` (6011 passed, 173 skipped)

## AI disclosure

GitHub Copilot helped identify edge cases and review the implementation. I reproduced the failures and validated the final change with targeted and full test suites.

### CODE REVIEW

File: src/specify_cli/workflows/catalog.py [hunk 1]
```diff
@@ -574,7 +578,9 @@ def _validate_redirect(_old_url: str, new_url: str) -> None:
             if cache_file.exists():
                 try:
                     with open(cache_file, encoding="utf-8") as f:
-                        return json.load(f)
+                        cached = json.load(f)
+                    if isinstance(cached, dict):
+                        return cached
```
[the remark this answers is not in the document]
Comment (marcelsafin) [reply src/specify_cli/workflows/catalog.py:583]: Added `test_non_mapping_stale_workflow_catalog_is_rejected`. It forces `open_url` to fail and verifies a non-object stale cache raises `WorkflowCatalogError` instead of being returned. Verified with the workflow test file, pinned Ruff, and the full suite (6,012 passed, 173 skipped).

AI disclosure: GitHub Copilot helped draft this response.

**Comment by marcelsafin** [comment]
@copilot review

**Comment by mnriem** [comment]
Thank you!
